### ISSUE TITLE
Problem with local transitions

### ISSUE THREAD

**Issue by lmatteis** [davidkpiano/xstate#71]
If I have this statechart:

```js
{
  initial: 'Hidden',
  states: {
    Hidden: {
      on: {
        PUBLISH_FAILURE: 'Failure',
      },
    },
    Failure: {
      on: {
        CLICKED_CLOSE: 'Hidden',
      },
    },
  },
}
```

When I'm in 'Failure' and the `CLICKED_CLOSE` event is fired, I properly go into the `Hidden` state. The problem is that I want to put the `CLICKED_CLOSE` event one level higher:

```js
{
  initial: 'Hidden',
  on: {
    CLICKED_CLOSE: '.Hidden',
  },
  states: {
    Hidden: {
      on: {
        PUBLISH_FAILURE: 'Failure',
      },
    },
    Failure: {
    },
  },
}
```

When I do this, the transition doesn't happen. I even tried doing `CLICKED_CLOSE: 'Alert.Hidden',` -- where Alert is the name of the parent state.

Any idea what I'm doing wrong?

**Comment by davidkpiano** [comment]
Ah yeah, this might be a bug. I'll look into it.

**Comment by mogsie** [comment]
This seems to be a problem with events declared at the top level. I added a couple of test cases, one where the `on CLICKED_CLOSED` was declared at the top level, the other in an `outer` top level state; otherwise identical statecharts. Only the first one fails.  Digging to see if there's a general problem with top level states, or if it only has to do with local transitions.

**Comment by mogsie** [comment]
Further testing seems to indicate that it's simply top level machine events that don't seem to work.  I moved the failing test to `machine.test.ts` instead.

**Comment by mogsie** [comment]
It fails in the `scxml` branch too.

**Comment by AnimaLupi** [comment]
Hello, sorry to bother you guys, but the fix is going to be backported to the incoming 3.2 release too?
And by the way, is there any approximative release date for 3.2?
Thank you!

**Comment by davidkpiano** [comment]
@AnimaLupi This will be fixed into 3.2, which I plan on releasing at React Finland (April 24-26).

**Comment by AnimaLupi** [comment]
@davidkpiano Thank you very much for the release info and fast reply! Nice work by the way!
